insights/client/archive.py

```diff
@@ -223,12 +226,40 @@ def cleanup_tmp(self):
             and tar_file exists.
         '''
         if self.config.keep_archive and self.tar_file:
+            self.storing_archive()
             if self.config.no_upload:
-                logger.info('Archive saved at %s', self.tar_file)
+                logger.info('Archive saved at %s', self.archive_stored)
             else:
-                logger.info('Insights archive retained in %s', self.tar_file)
-            if self.config.obfuscate:
-                return  # return before deleting tmp_dir
-        else:
-            self.delete_archive_file()
+                logger.info('Insights archive retained in %s', self.archive_stored)
         self.delete_tmp_dir()
+
+    def sigterm_handler(_signo, _stack_frame, _context):
+        sys.exit(1)
+
+    def cleanup_previous_archive(self):
+        '''
+        Used at the start, this will clean the temporary directory of previous killed runs
+        '''
+        for file in glob.glob('/var/tmp/insights-archive-*'):
+            os.path.join('', file)
+            logger.debug("Deleting previous archive %s", file)
+            shutil.rmtree(file, True)
+
+    def storing_archive(self):
+        keep_archive_dir = '/var/cache/insights-client'
+        if not os.path.exists(keep_archive_dir):
+            try:
+                os.makedirs(keep_archive_dir)
+            except OSError as e:
+                logger.error(e)
```

Comment:
We often get bug reports complaining about stack traces in console output. Maybe we can change this line to `logger.debug`? Does printing the exception here print the entire stack trace or just the exception's class & message?

Reply:
It prints the stack trace but it was a leftover of debugging it, also I have noticed the test wasn't good for this method